packages/hardhat-core/src/internal/context.ts

```diff
@@ -78,16 +78,25 @@ export class HardhatContext {
   }
 
   public getFilesLoadedDuringConfig(): string[] {
```

Comment:
This is an unrelated concept. What we want to check is the `userConfig` here: https://github.com/nomiclabs/hardhat/blob/master/packages/hardhat-core/src/internal/core/config/config-loading.ts#L66